MobileCenterAnalytics/MobileCenterAnalytics/MSAnalytics.m

```diff
@@ -146,32 +154,90 @@ + (BOOL)isAutoPageTrackingEnabled {
 
 #pragma mark - Private methods
 
-- (BOOL)validateProperties:(NSDictionary<NSString *, NSString *> *)properties {
+- (BOOL)validateEventName:(NSString *)eventName {
+  if (!eventName) {
+    MSLogError([MSAnalytics logTag],
+               @"The event name is null. It can not be null.");
+    return NO;
+  }
+  if ([eventName length] < minEventNameLength) {
+    MSLogError([MSAnalytics logTag],
+               @"The event name is too short. It should be more then %d", minEventNameLength);
+    return NO;
+  }
+  if ([eventName length] > maxEventNameLength) {
+    MSLogError([MSAnalytics logTag],
+               @"The event name is too long. It should be less then %d", maxEventNameLength);
+    return NO;
+  }
+  return YES;
+}
+
+- (NSDictionary<NSString *, NSString *> *)validateProperties:(NSDictionary<NSString *, NSString *> *)properties {
+  if([properties count] > maxPropertriesPerEvent) {
+    MSLogWarning([MSAnalytics logTag],
+                 @"The log contains too many properties. Only first %d valid properties will be send.",
+                 maxPropertriesPerEvent);
+  }
+  int totalValidProps = 0;
+  NSMutableDictionary<NSString *, NSString *> *validProperties = [NSMutableDictionary new];
   for (id key in properties) {
     if (![key isKindOfClass:[NSString class]] || ![properties[key] isKindOfClass:[NSString class]]) {
-      return NO;
+      continue;
+    }
+
+    // Validate key
+    NSString *strKey = key;
+    if (!strKey ||
+        [strKey length] < minPropertyKeyLength ||
+        [strKey length] > maxPropertyKeyLength) {
+      MSLogWarning([MSAnalytics logTag],
+                   @"The properties contain invalid key %@. Property will be skipped.",
+                   strKey);
+      continue;
+    }
+
+    // Validate value
+    NSString *value = properties[key];
+    if (!value) {
+      MSLogWarning([MSAnalytics logTag],
+                   @"The properties contain invalid value %@. Property will be skipped.",
+                   value);
+      continue;
+    }
+    if(value && [value length] > maxPropertyValueLength) {
+      MSLogWarning([MSAnalytics logTag],
+                   @"The properties contain invalid value %@. Property will be skipped.",
+                   value);
+      continue;
+    }
+
+    // Save valid properties
+    [validProperties setObject:value forKey:key];
+    if (++totalValidProps == maxPropertriesPerEvent) {
+      break;
     }
```

Comment:
@TroubleMakerBen I can remove `totalValidProps` and use a dictionary's property `count`. I think it will be more obvious and more clever. `totalValidProps` is excess variable.